evm-contracts/src/v0.6-ovm/AccessControlledAggregator.sol

```diff
@@ -1,217 +1,111 @@
-   * @return answeredInRound is the round ID of the round in which the answer
-   * was computed. answeredInRound may be smaller than roundId when the round
-   * timed out. answerInRound is equal to roundId when the round didn't time out
-   * and was completed regularly.
-   * @dev overridden funcion to add the checkAccess() modifier
-   * @dev Note that for in-progress rounds (i.e. rounds that haven't yet
-   * received maxSubmissions) answer and updatedAt may change between queries.
+   * @return The Address of the implementation.
    */
-  function latestRoundData()
-    public
-    view
-    override
-    checkAccess()
-    returns (
-      uint80 roundId,
-      int256 answer,
-      uint256 startedAt,
-      uint256 updatedAt,
-      uint80 answeredInRound
-    )
-  {
-    return super.latestRoundData();
+  function _implementation() internal view  override returns (address) {
+    return implementation;
   }
 
   /**
-   * @notice get the most recently reported answer
-   * @dev overridden funcion to add the checkAccess() modifier
-   *
-   * @dev #[deprecated] Use latestRoundData instead. This does not error if no
-   * answer has been reached, it will simply return 0. Either wait to point to
-   * an already answered Aggregator or use the recommended latestRoundData
-   * instead which includes better verification information.
+   * @dev Function that is run as the first thing in the fallback function.
+   * Can be redefined in derived contracts to add functionality.
+   * Redefinitions must call super._willFallback().
    */
-  function latestAnswer()
-    public
-    view
-    override
-    checkAccess()
-    returns (int256)
-  {
-    return super.latestAnswer();
+  function _willFallback() internal override virtual {
+    super._willFallback();
+    // Check if one of the guarded functions
+    if (guardedFunctions[msg.sig]) {
+      require(hasAccess(msg.sender, msg.data), "No access");
+    }
   }
 
   /**
-   * @notice get the most recently reported round ID
-   * @dev overridden funcion to add the checkAccess() modifier
-   *
-   * @dev #[deprecated] Use latestRoundData instead. This does not error if no
-   * answer has been reached, it will simply return 0. Either wait to point to
-   * an already answered Aggregator or use the recommended latestRoundData
-   * instead which includes better verification information.
+   * @dev Storage slots for the Owned contract storage.
+   * As we expect the implementation contract to extend both Initializable ([0,50] slots reserved) and Owned,
+   * and this proxy contract is also Owned, we need to make sure that Owned storage in this contract
+   * alligns with Owned storage in the underlying contract.
    */
-  function latestRound()
-    public
-    view
-    override
-    checkAccess()
-    returns (uint256)
-  {
-    return super.latestRound();
-  }
+  uint8 internal constant SLOT_OWNER = 51;
+  uint8 internal constant SLOT_PENDIG_OWNER = SLOT_OWNER + 1;
```

Comment:
This seems brittle and the contract is substantially smaller now. Can we not just inherit from Owned here?

Reply:
As discussed on the call, needs to align with `Owned` storage in FA contract (underlying implementation), so both codebases (ACA & FA) have `Owned` storage in the same location and behave as expected.

I have some ideas about simplifying storage setup, so I'll investigate that, but not sure it will touch this `Owned` complexity without major changes.